### PR TITLE
treewide: rename name to pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Description of changes
#103997

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/bitcoin-unlimited/default.nix [hunk 1]
```diff
@@ -6,7 +6,7 @@
 with lib;
 
 stdenv.mkDerivation rec {
-  name = "bitcoin" + (toString (optional (!withGui) "d")) + "-unlimited-" + version;
+  pname = "bitcoin" + (toString (optional (!withGui) "d")) + "-unlimited";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/bitcoin-unlimited/default.nix:9]:
```suggestion
  pname = "bitcoin" + optionalString (!withGui) "d" + "-unlimited";
```

File: pkgs/applications/blockchains/dogecoin/default.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 with lib;
 stdenv.mkDerivation rec {
-  name = "dogecoin" + (toString (optional (!withGui) "d")) + "-" + version;
+  pname = "dogecoin" + (toString (optional (!withGui) "d"));
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/dogecoin/default.nix:9]:
```suggestion
  pname = "dogecoin" + optionalString (!withGui) "d";
```

File: pkgs/applications/blockchains/litecoin/default.nix [hunk 3]
```diff
@@ -11,8 +11,7 @@
 with lib;
 
 mkDerivation rec {
-
-  name = "litecoin" + (toString (optional (!withGui) "d")) + "-" + version;
+  pname = "litecoin" + (toString (optional (!withGui) "d"));
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/litecoin/default.nix:14]:
```suggestion
  pname = "litecoin" + optionalString (!withGui) "d";
```

File: pkgs/applications/blockchains/namecoin/default.nix [hunk 4]
```diff
@@ -4,7 +4,7 @@
 with lib;
 stdenv.mkDerivation rec {
   version = "nc22.0";
-  name = "namecoin" + toString (optional (!withGui) "d") + "-" + version;
+  pname = "namecoin" + toString (optional (!withGui) "d");
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/namecoin/default.nix:7]:
```suggestion
  pname = "namecoin" + optionalString (!withGui) "d";
```

File: pkgs/applications/editors/jetbrains/linux.nix [hunk 5]
```diff
@@ -3,20 +3,20 @@
...
 
-{ name, product, productShort ? product, version, src, wmClass, jdk, meta, extraLdPath ? [], extraWrapperArgs ? [] }@args:
+{ pname, product, productShort ? product, version, src, wmClass, jdk, meta, extraLdPath ? [], extraWrapperArgs ? [] }@args:
 
 with lib;
 
 let loName = toLower productShort;
     hiName = toUpper productShort;
-    mainProgram = concatStringsSep "-" (init (splitString "-" name));
+    mainProgram = concatStringsSep "-" (init (splitString "-" pname));
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/jetbrains/linux.nix:12]:
```suggestion
```
mainProgram looks like it was pname already or am I missing something?

File: pkgs/applications/emulators/wine/base.nix [hunk 6]
```diff
@@ -42,9 +42,9 @@ stdenv.mkDerivation ((lib.optionalAttrs (buildScript != null) {
     make loader/wine64-preloader NIX_LDFLAGS="" NIX_LDFLAGS_${stdenv.cc.suffixSalt}=""
   '';
 }) // rec {
-  inherit src;
+  inherit version src;
 
-  name = if supportFlags.waylandSupport then "${prevName}-wayland" else prevName;
+  pname = if supportFlags.waylandSupport then "${prevName}-wayland" else prevName;
```
Comment (SuperSandro2000) [inline pkgs/applications/emulators/wine/base.nix:47]:
```suggestion
  pname = prevName + lib.optionalString supportFlags.waylandSupport "wayland";
```

File: pkgs/applications/editors/jetbrains/linux.nix (hunk 5 above)
Comment (Stunkymonkey) [reply pkgs/applications/editors/jetbrains/linux.nix]: fine this way?

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 165303` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>bitcoin-unlimited</li>
    <li>bitcoind-unlimited</li>
    <li>gnuradio3_8</li>
    <li>gnuradio3_9</li>
    <li>litecoin</li>
    <li>litecoind</li>
    <li>qxmledit</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>cardinal</li>
    <li>grass</li>
    <li>wine64Packages.staging</li>
    <li>wine64Packages.stagingFull</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>inkscape-extensions.hexmap</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
```
/nix/store/rpk8vd0habrxqfj7mg29yk56gsyvg1b1-clion-2021.3.3/clion/plugins/remote-dev-server/bin/launcher.sh: interpreter directive changed from "#!/bin/sh" to "/nix/store/dndi916j6yxzfzzj2sma2llhrlwahq06-bash-5.1-p16/bin/sh"
checking for references to /build/ in /nix/store/rpk8vd0habrxqfj7mg29yk56gsyvg1b1-clion-2021.3.3...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
/nix/store/lkk0hf9h4x85mb4lrsf2sh25qm2i4n18-stdenv-linux/setup: line 87: cd: /nix/store/rpk8vd0habrxqfj7mg29yk56gsyvg1b1-clion-2021.3.3/clion-2021.3.3: No such file or directory
error: builder for '/nix/store/ywqpaza1v0wpgvfhvp8a54y8j24kvnbw-clion-2021.3.3.drv' failed with exit code 1;
```

This update broke clion for me. Is anybody testing these changes before they go in?

**Comment by mweinelt** [comment]
The CLion breakage is now tracked in #166216.

**Pull request merged**